### PR TITLE
Update NC-fluxes for the SWE-Wet/Dry

### PR DESCRIPTION
**Pull request by patrickersing**
In https://github.com/trixi-framework/Trixi.jl/pull/2038, the definition of `flux_nonconservative_wintermeyer_etal` has been changed and the method `flux_nonconservative_ersing_etal` is removed from `Trixi.jl`.

Therefore in this PR `flux_nonconservative_ersing_etal` is now exported by `TrixiShallowWater.jl` and some test values had to be updated. Since `flux_nonconservative_ersing_etal` moves to TrixiShallowWater.jl the lower compat bound for `Trixi.jl` is also updated.

Needs to be merged after  https://github.com/trixi-framework/Trixi.jl/pull/2038

### CODE REVIEW

**Comment by patrickersing** [comment]
> The changes to code look good and it all should work. Because the CI will not run until Trixi v0.8.7 is released did you run the tests locally that everything passes?

Yes, I checked that the tests pass locally.
Since we need to merge the PR from Trixi.jl first, I would also wait for the CI to pass before merging to see that everything really works as expected.

**Comment by ranocha** [comment]
I'll re-open this to trigger CI again using the new release of Trixi.jl

**Comment by ranocha** [comment]
Thanks! Looks like it works. It would be great to get a new release including this somewhat soon so that Trixi.jl downstream CI passes again.

**Review by andrewwinters5000 — APPROVED** [review]

**Pull request merged**
